fitbenchmarking/fitbenchmarking
# 260 Refactor options

#### Description of Work

Fixes #260 

Refactors the options and changes the input format to a .ini file.
The .ini file improves on the json version as it allows comments, and is much more human readable.

The options are now implemented as a class, and self document based on the default values.
This format allows a priority which is consistent by design rather than convention and is as follows:
1. Values set in code
2. Values set in options file
3. Values predefined in the options class (default)


#### Testing Instructions

1. Run the code with varying options and check that results are as expected
2. Read through the docs [here](https://fitbenchmarking.readthedocs.io/en/260-options_refactor/users/options.html) to check that they make sense.


Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
